README.md

```diff
@@ -1,2 +1,22 @@
 # comp110-worksheet-C
-Base repository for COMP110 worksheet C
+Base repository for COMP110 worksheet C - Aidan Crossan
+
+**a)** The algorithm performs a search to see if there are duplicate items in the same list (e.g. to see if there are more than two people in a list who have the name 'Garry' for instance).
+
+**b)** The least efficient way of cycling through the list is to check each of the items in the list in pairs to see whether they match or not. This would be quadratic complexity O(n^2).
+
+**c)** i-1 will still be similar to n-1, it just means that the j loop will be one behind the i loop. 
+
+**d)** The two loops ran twice in the original version meaning it had to go through one list and then run it again to see if there was any overlap.
+
+**e)** The time complexity is still quadratic as the task the algorithm performs is still the same. It's roughly twice as fast because it cuts down the number of searches by about a quarter.
```

Comment:
Two algorithms can perform the same task but have different time complexities! It is indeed still quadratic -- because speeding up an algorithm by a constant factor doesn't change its time complexity